lib/middlewares/src/metering.rs

```diff
@@ -173,14 +211,28 @@ impl<F: Fn(&Operator) -> u64 + Copy + Clone + Send + Sync> FunctionMiddleware
 ///
 /// The instance Module must have been processed with the [`Metering`] middleware
 /// at compile time, otherwise this will panic.
-pub fn get_remaining_points(instance: &Instance) -> u64 {
-    instance
+pub fn get_remaining_points(instance: &Instance) -> MeteringPoints {
+    let points = instance
         .exports
-        .get_global("remaining_points")
-        .expect("Can't get `remaining_points` from Instance")
+        .get_global("wasmer_metering_remaining_points")
+        .expect("Can't get `wasmer_metering_remaining_points` from Instance")
         .get()
         .try_into()
-        .expect("`remaining_points` from Instance has wrong type")
+        .expect("`wasmer_metering_remaining_points` from Instance has wrong type");
+
+    let error: u64 = instance
+        .exports
+        .get_global("wasmer_metering_missing_points")
+        .expect("Can't get `wasmer_metering_missing_points` from Instance")
+        .get()
+        .try_into()
+        .expect("`wasmer_metering_missing_points` from Instance has wrong type");
+
+    if error > 0 {
+        return MeteringPoints::Depleted(points, error);
```

Comment:
With this new higher level interface I don't think we need the error code anymore.
I think the `points` are not interesting in the `Depleted` case as well.